### DISCUSSION TITLE
First Roadmap

### DISCUSSION THREAD

**Discussion by Aaronontheweb** [AfterLutz/Akkatecture#33] [General]
The first thing that Akkatecture needs in order to get going again is a roadmap - what problems do you want to solve and how do you want to solve them? Start filing some issues, creating projects, and creating some milestones. These help contributors who want to help organize their participation into actual releases.

Maybe begin by porting some of the open issues from https://github.com/Lutando/Akkatecture/issues ?

**Comment by malclear** [comment]
Now that the Akkatecture libraries have been completely rehomed under the AfterLutz organization, including ownership of the akkatecture.net web domain and the ability to publish to NuGet with the original names, goals for ongoing development should be clearly defined. I'm thinking that some goals for the continued development of Akkatecture should include the following (in no particular order): 

- develop a new library to support resumable projections
- utilize the new Akka.Hosting library in all our projects where appropriate
- migrate to Akka.NET v1.5 when it's available later this year
- build on the documentation, guidance and examples available in both code and on the website

Please comment your thoughts in this Discussion. After we've had time to deliberate, I will update the website with the final version.

**Reply by jblackburn21** [reply]
Currently, the default JSON serialization is used for messaging and persistence. This makes it simple to get started, but introduces challenges in a production event sourced system since messages and state evolve over time.

This project should look into providing guidance on using Protofbuf for serialization, or adding abstractions to simplify the serialization configuration.
